File: ccif/src/java/other/org/himalay/ccif/RequestPidsCnf.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:33:09 EST 2016")

public  class RequestPidsCnf extends   APDUBaseFactory.APDUBase { //Concrete type is RequestPidsCnf




    // members variables
    // header
    public APDUHeader header ;
    // length
    public short length ;
    // ltsid
    public short ltsid ;
    // numOfPids
    public ArrayList<pids> numOfPids ;

    public RequestPidsCnf () // throws Exception
    {
        init();
    }

    private void init()
    {
        // Initialize header
        header = new APDUHeader();
        // Initialize length

        // Initialize ltsid

        // Initialize numOfPids
        numOfPids= new ArrayList<pids>();
        numOfPids.setMemberSize(0);
    }

    public int readNoHeader(DataInputStream istream) throws IOException
    {

        preRead();
        int retVal= 0;
        // read length
        {
            length=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read ltsid
        {
            ltsid=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read numOfPids
        for (; istream.available() > 0 ; ) {
            pids temp;
            temp = new pids();
            retVal += temp.read(istream);
            numOfPids.add(temp);
        }

        postRead();
        return retVal;
    }

    public int read(DataInputStream istream) throws IOException
    {
        preRead();
        int retVal= 0;

        // read header
        retVal += header.read(istream);
        // read length
        {
            length=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read ltsid
        {
            ltsid=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read numOfPids
        for (; istream.available() > 0 ; ) {
            pids temp;
            temp = new pids();
            retVal += temp.read(istream);
            numOfPids.add(temp);
        }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException
    {
        preWrite();
        int retVal= 0;

        {   /** fix dependent sizes for header **/
        }


        {   /** fix dependent sizes for numOfPids**/
        }

        // write header
        if (header!=null)retVal +=header.write(ostream);
        // write length
        ostream.writeByte(length);
        retVal +=1;
        // write ltsid
        ostream.writeByte(ltsid);
        retVal +=1;
        // write numOfPids
        {
            ArrayList<pids> temp1 = numOfPids;
            for (int iIdx=0; iIdx < temp1.getCount() ; iIdx++) {
                pids temp2    =    temp1.get(iIdx);
                if (temp2!=null)retVal +=temp2.write(ostream);
            }
        }
        postWrite();
        return retVal;
    }

    public int dump(DumpContext dc) throws IOException
    {
        dc.indent();
        dc.getPs().print("RequestPidsCnf\n");
        dc.increaseIndent();
        int retVal= 0;
        // write header
        if ( header != null ) {
            dc.indent();
            dc.getPs().println("header") ;
            retVal +=header.dump(dc);
        }
        // write length
        dc.indent();
        dc.getPs().println("length="+length+"(0x"+ Integer.toHexString(length) + ")" );
        // write ltsid
        dc.indent();
        dc.getPs().println("ltsid="+ltsid+"(0x"+ Integer.toHexString(ltsid) + ")" );
        // write numOfPids
        {
            ArrayList<pids> temp1 = numOfPids;
            for (int iIdx=0; iIdx < temp1.getCount() ; iIdx++) {
                pids element    = temp1.get(iIdx);
                dc.indent();
                dc.getPs().println(iIdx);
                if ( element != null ) {
                    dc.indent();
                    dc.getPs().println("element") ;
                    retVal +=element.dump(dc);
                }
            }
        }
        dc.decreaseIndent();
        return retVal;
    }


    // Getter for header
    //public APDUHeader getHeader()
    //{
    //    return header ;
    //}


    // Setter for header
    //public void setHeader(APDUHeader val)
    //{
    //    this.header= val;
    //}
    // Getter for length
    //public short getLength()
    //{
    //    return length ;
    //}


    // Setter for length
    //public void setLength(short val)
    //{
    //    this.length= val;
    //}
    // Getter for ltsid
    //public short getLtsid()
    //{
    //    return ltsid ;
    //}


    // Setter for ltsid
    //public void setLtsid(short val)
    //{
    //    this.ltsid= val;
    //}
    // Getter for numOfPids
    //public ArrayList<pids> getNumOfPids()
    //{
    //    return numOfPids ;
    //}


    // Setter for numOfPids
    //public void setNumOfPids(ArrayList<pids> val)
    //{
    //    this.numOfPids= val;
    //}


    public int addToNumOfPids(pids val)
    {
        numOfPids.add(val);
        return numOfPids.size();
    }


    public int removeFromNumOfPids(pids val)
    {
        numOfPids.remove(val);
        return numOfPids.size();
    }


    public int removeNthFromNumOfPids(int idx)
    {
        numOfPids.remove(idx);
        return numOfPids.size();
    }


    public int emptyNumOfPids(int idx)
    {
        numOfPids.clear();
        return numOfPids.size();
    }


    public int getSize() throws IOException
    {
        DataOutputStream dos= new DataOutputStream(new NullStream());
        return this.write(dos);
    }


    public void setHeader(APDUHeader header)
    {
        this.header= header;
    }

    public APDUHeader getHeader()
    {
        return this.header;
    }

}

// End of code

File: libccif/src/bp/RequestPidsCnf.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:18 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "RequestPidsCnf.h"

#ifdef org_himalay_ccif_RequestPidsCnf__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
RequestPidsCnf::RequestPidsCnf() {
    references= 0;


    // header
    org_himalay_ccif_APDUHeader__NEW( header);
    // length
    length= (ui8)0;
    // ltsid
    ltsid= (ui8)0;
    // numOfPids
    numOfPids.setMemberSize(0);
}
// Destructor
RequestPidsCnf::~RequestPidsCnf() {
    // header
    org_himalay_ccif_APDUHeader__DELETE( header);
    // length

    // ltsid

    // numOfPids
    {
#ifndef org_himalay_ccif_pids_USE_SMART_PTR
        numOfPids.deleteMembers();
#endif
    }
}

int RequestPidsCnf::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // length
    {
        length=(istream.readUI8());
        retVal+=1;
    }
    // ltsid
    {
        ltsid=(istream.readUI8());
        retVal+=1;
    }
    // numOfPids
    for (; istream.available() > 0 ; ) {
        org_himalay_ccif_pids__PTR_TYPE temp;
        org_himalay_ccif_pids__NEW( temp);
        retVal += temp->read(istream);
        numOfPids.add(temp);
    }
    return retVal;
}

int RequestPidsCnf::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read header
    retVal += header->read(istream);
    // read length
    {
        length=(istream.readUI8());
        retVal+=1;
    }
    // read ltsid
    {
        ltsid=(istream.readUI8());
        retVal+=1;
    }
    // read numOfPids
    for (; istream.available() > 0 ; ) {
        org_himalay_ccif_pids__PTR_TYPE temp;
        org_himalay_ccif_pids__NEW( temp);
        retVal += temp->read(istream);
        numOfPids.add(temp);
    }

    return retVal;
}

int RequestPidsCnf::write(DataOutputStream& ostream) { // throws IOException
    {   /** fix dependent sizes for header **/
    }


    {   /** fix dependent sizes for numOfPids**/
    }

    int retVal= 0;
    // write header
    if ( header != NULL ) {
        retVal +=header->write(ostream);
    }
    // write length
    ostream.writeUI8(length);
    retVal +=1;
    // write ltsid
    ostream.writeUI8(ltsid);
    retVal +=1;
    // write numOfPids
{   for (ArrayList_iterator(pids) it=  numOfPids.getIterator() ; it.hasNext();) {
            {
                retVal +=it.get()->write(ostream);
            }
        }
    }

    return retVal;
}

int RequestPidsCnf::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("RequestPidsCnf\n");
    dc.increaseIndent();
    int retVal= 0;
    // write header
    if ( header != NULL ) {
        dc.indent();
        dc.getPs().println("header") ;
        retVal +=header->dump(dc);
    }
    // write length
    dc.indent();
    dc.getPs().print("length=");
    dc.getPs().println((long)length);
    // write ltsid
    dc.indent();
    dc.getPs().print("ltsid=");
    dc.getPs().println((long)ltsid);
    // write numOfPids
{   for (ArrayList_iterator(pids) it=  numOfPids.getIterator() ; it.hasNext();) {
            {
                dc.indent();
                dc.getPs().println("element");
                retVal +=it.get()->dump(dc);
            }
        }
    }
    dc.decreaseIndent();
    return retVal;
}


// Getter for header
//org_himalay_ccif_APDUHeader__PTR_TYPE RequestPidsCnf::getHeader()
//{
//return this->header;
//}

// Setter for header
//void RequestPidsCnf::setHeader(org_himalay_ccif_APDUHeader__PTR_TYPE val)
//{
//this->header= val;
//}

// Getter for length
//ui8 RequestPidsCnf::getLength()
//{
//return this->length;
//}

// Setter for length
//void RequestPidsCnf::setLength(ui8 val)
//{
//this->length= val;
//}

// Getter for ltsid
//ui8 RequestPidsCnf::getLtsid()
//{
//return this->ltsid;
//}

// Setter for ltsid
//void RequestPidsCnf::setLtsid(ui8 val)
//{
//this->ltsid= val;
//}

// Getter for numOfPids
//ArrayList(pids) RequestPidsCnf::getNumOfPids()
//{
//return this->numOfPids;
//}

// Setter for numOfPids
//void RequestPidsCnf::setNumOfPids(ArrayList(pids) val)
//{
//this->numOfPids= val;
//}

int RequestPidsCnf::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

void RequestPidsCnf::setHeader(org_himalay_ccif_APDUHeader__PTR_TYPE header)
{
    this->header= header;
}

org_himalay_ccif_APDUHeader__PTR_TYPE RequestPidsCnf::getHeader()
{
    return this->header;
}

#ifdef org_himalay_ccif_RequestPidsCnf__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::RequestPidsCnf* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::RequestPidsCnf* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code